### PR TITLE
docs: Update triage process description

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Updates the issue triage process to discuss the new Priority and Impact fields.

#### Is there anything you'd like reviewers to focus on?

Does the description make sense?

### CODE REVIEW

File: docs/src/maintain/manage-issues.md [hunk 1]
```diff
@@ -64,14 +64,24 @@ The steps for triaging an issue are:
...
+        * Please add labels describing the part of ESLint affected:
             * **3rd party plugin**: Related to third-party functionality (plugins, parsers, rules, etc.)
             * **build**: Related to commands run during a build (testing, linting, release scripts, etc.)
             * **cli**: Related to command line input or output, or to `CLIEngine`
             * **core**: Related to internal APIs
             * **documentation**: Related to content on eslint.org
             * **infrastructure**: Related to resources needed for builds or deployment (VMs, CI tools, bots, etc.)
             * **rule**: Related to core rules
+        * Please assign an initial priority based on the importance of the issue. If you're not sure, use your best judgment. We can always change the priority later.
+            * **P1**: Urgent and important, we need to address this immediately.
```
Comment (snitin315) [inline docs/src/maintain/manage-issues.md:76]: How about using `P0` for high priority?

**Review by aladdin-add — COMMENTED** [review]
can we add some common examples - such as only a certain rule-related feature request/bug should belong to which level, this can help us better unify the standard. :)

File: docs/src/maintain/manage-issues.md (hunk 1 above)
Comment (nzakas) [reply docs/src/maintain/manage-issues.md:76]: I think P1 is fine for this use case.

**Comment by nzakas** [comment]
@aladdin-add I think we need to run with the process a bit to figure out specific examples.

**Review by aladdin-add — APPROVED** [review]
LGTM. waiting @mdjermanovic and other members to review before merging.

**Review by mdjermanovic — APPROVED** [review]

**Comment by snitin315** [comment]
Re-submitted in https://github.com/eslint/eslint/pull/17266

**Pull request merged**